### DISCUSSION TITLE
Handling API errors

### DISCUSSION THREAD

**Discussion by avalladaresm** [TanStack/query#1101] [General]
Is there a specific way to handle error messages from Expressjs for example?
Say I do `res.status(404).send({message:'this is the error message'})`, is there a way to get that message on the onError callback using the useMutation hook? All I get from the error is a default error message `Request failed with status code 404`.

**Comment by TkDodo** [comment]
Can you create a codesandbox example maybe? Happy to look into it 😊

**Reply by avalladaresm** [reply]
Well... It's actually a bit hard to explain now... I was just finishing the codesandbox example to post it here and figured it out by looking it [here](https://stackoverflow.com/questions/41854793/axios-cannot-access-message-of-an-error). It seems it wasn't a react-query issue, more like me not knowing how to exactly get the custom error message from axios, in which case, the way to get it on the `onError` callback on the `useMutation` hook would be 
```
onError: (err, variables) => {
    console.log("err", err.response.data.message);
}
```
and what I was doing was actually `console.log("err", err.message)`, that's why I only received `Request failed with status code 404` as stated above.

Thanks for wanting to help! :smiley:
